Automattic/newspack-ios
# Issues/58 folder data reconciliation final

Closes #58 

This is the final PR in the series of file system / core data reconciliation patches for Site and StoryFolders. Another round will follow for StoryFolder contents in the future. 

This PR introduces a Reconciler that checks for inconsistencies between the file system and core data whenever a new session is started, or when the app becomes active.  If inconsistencies are detected the Reconciler attempts to remove items from core data that no longer have counterparts in the file system, and creates new entries in core data for anything discovered in the filesystem.

To test:
- As always, make sure all unit tests pass.  The new unit tests for the Reconciler can take a moment or two.

Test Case:
- Open the app and create a bunch of StoryFolders. 
- Background the app and open the Files app. 
- Find the Newspack contents, and find the folders for the StoryFolders you created.
- Delete a few. 
- Long press to get a dialog to create a new folder. Do so, and give the new folder a name. 
- Return to Newspack and check that the list of story folders updates to reflect the file system changes. 

@jleandroperez are you game for a Newspack PR?
